nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/AsyncService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nelson Silva <nsilva@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.api;
+
+import java.io.Serializable;
+
+/**
+ * Interface to be implemented by asynchronous services.
+ *
+ * @param <K> type of command id
+ * @param <S> type of status
+ * @param <V> type of result
+ * @since 10.3
+ */
+public interface AsyncService<K extends Serializable, S, V> {
+
+    /**
+     * Returns the status of a command.
+     */
+    S getStatus(K commandId);
+
+    /**
+     * Aborts a scheduled or running command. All computations related to this command will be skipped.
+     */
+    S abort(K commandId);
+
+    /**
+     * Returns true is command is completed.
+     */
+    boolean isCompleted(K commandId);
```

Comment:
Could we define and add a state on AsyncStatus and use it ?
If we put isCompleted here, others will follow isCancelled, isFailed, etc.

Reply:
make sense yes

Reply:
At this point I really only need to know whether it's completed or not. Perhaps we can review this as part of https://jira.nuxeo.com/browse/NXP-26246 since we'll need proper error states